core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1498,25 +1498,25 @@ class UnifiedLog(@volatile var logStartOffset: Long,
     producerStateManager.takeSnapshot()
     updateHighWatermarkWithLogEndOffset()
     // Schedule an asynchronous flush of the old segment
-    scheduler.schedule("flush-log", () => flush(newSegment.baseOffset))
+    scheduler.schedule("flush-log", () => flushUptoOffsetExclusive(newSegment.baseOffset))
     newSegment
   }
 
   /**
    * Flush all local log segments
    *
-   * @param inclusive should be true during a clean shutdown, and false otherwise. The reason is that
+   * @param forceFlushActiveSegment should be true during a clean shutdown, and false otherwise. The reason is that
    * we have to pass logEngOffset + 1 to the `localLog.flush(offset: Long): Unit` function to flush empty
    * active segments, which is important to make sure we don't lose the empty index file during shutdown.
```

Comment:
IIUC the main reason we would like to force flush the active segment is slightly different.
Instead of `which is important to make sure we don't lose the empty index file during shutdown.`, could the comment say `which is important to make sure we persist the active segment file during shutdown, particularly when its empty.` ?